bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java

```diff
@@ -134,6 +135,18 @@ public void setLabel(@Nullable String label) {
         return Collections.unmodifiableList(new ArrayList<>(this.channels));
     }
 
+    @Override
+    public List<Channel> getChannelsOfGroup(String channelGroupId) {
+        List<Channel> channels = new ArrayList<>();
+        for (Channel channel : this.channels) {
```

Comment:
We can use the Java streaming library for this approach too. But then we have to change the property channels to use a thread safe list (e.g. CopyOnWriteArrayList). Does that make sense?